### PR TITLE
[WIP] Add client package for opensds controller project

### PR DESCRIPTION
**Pull request by leonwanghui**
This patch targets as issue [#102](https://github.com/opensds/opensds/issues/102), more info can be found in README.md.

More suggestions are welcomed.

### CODE REVIEW

**Comment by wisererik** [comment]
@leonwanghui Nice job, so the [nbp](https://github.com/opensds/nbp) can use this library~

File: client/dock.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+	if err := d.Recv(request, url, "GET", d.Opt, &res); err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (d *DockMgr) ResetAndUpdateDockRequestContent(in interface{}) error {
```
Comment (xxwjj) [inline client/dock.go]: This method name  is too loog, it`s not good for reading, can you make a shorter one?

Comment (leonwanghui) [reply client/dock.go]: Ok, I'll have a try, thanks jerry!

**Comment by xxwjj** [comment]
@leonwanghui There are too points I think we need disscuss . 
1、One developer use the client funtcion  invoke tow method, IMO, it's too cumbersome to use.The fllowing usage maybe better:
```golang
package main

import (
	"fmt"
	
	"github.com/opensds/opensds/client"
	"github.com/opensds/opensds/pkg/model"
)

func main() {
	c := NewClient(&client.Config{
		Endpoint: ":8080",
	})
	
	fmt.Printf("Updated c is %v\n", &modelVolumeSpec{Name: "test"}))
	vol, err := c.CreateVolume()
	if err != nil {
		panic(err)
	}
}
```
2、For some simple scenarios, the parameters can not be encapsulated.For example the DeleteVolume funtcion, we just only send a volume of ID to it.

**Comment by leonwanghui** [comment]
@xxwjj Thanks for your suggestion, I have updated the patch and plz have a look.

**Pull request merged**